### PR TITLE
Random attempt at updating profile before joining

### LINKED ISSUE #24: Update profile before joining room
For the somewhat rare case of people bridging `irc <--> matrix <--> discord`. When the discord bridge joins the user to the matrix room it doesn't always have a display name set, so the IRC bridge picks up the user ID, leading to IRC users not having a clue who is speaking.

Blocking the join on a successful profile update would seem reasonable to me.

### PR DESCRIPTION
**Pull request by strk**
Hopefully fixes #24 (untested!)

\cc @TravisR (can you test this?)

### CODE REVIEW

File: src/discordbot.ts [hunk 1]
```diff
@@ -89,10 +89,7 @@ export class DiscordBot {
     let chan;
     let embed;
     const mxClient = this.bridge.getClientFactory().getClientAs();
-    return this.LookupRoom(guildId, channelId).then((channel) => {
-      chan = channel;
-      return mxClient.getProfileInfo(event.sender);
-    }).then((profile) => {
+    return this.LookupRoom(guildId, channelId).then((profile) => {
```
Comment (turt2live) [inline src/discordbot.ts]: The variables don't match here. `LookupRoom` resolves to a channel, not a profile.

**Comment by strk** [comment]
Is d231d99 any better ?

File: src/discordbot.ts [hunk 2]
```diff
@@ -89,10 +89,7 @@ export class DiscordBot {
     let chan;
     let embed;
     const mxClient = this.bridge.getClientFactory().getClientAs();
-    return this.LookupRoom(guildId, channelId).then((channel) => {
```
Comment (turt2live) [inline src/discordbot.ts]: I think this diff is actually in the wrong spot now that I've looked a bit closer at it. This is on the matrix -> discord side, not the discord -> matrix side (the flow causing issues).

**Comment by strk** [comment]
I hope you can look at this because I'm really just shooting in the
dark

**Comment by turt2live** [comment]
Looks like the fix has been PRd already: https://github.com/Half-Shot/matrix-appservice-discord/pull/8

**Comment by strk** [comment]
Closing this one then. Thanks for the heads up !

**Pull request closed without merging**